### ISSUE TITLE
Fix false negatives for custom properties within var() in custom-property-pattern

### ISSUE THREAD

**Issue by StephenEsser** [stylelint/stylelint#3655]
## Description
The `custom-property-pattern` rule is not enforced when applied inline to a property. The lint rule is only enforced when instantiated as a definition.

Many of our variables are defined using an inline fallback pattern so the variables can be themed by downstream applications. Example: `background-color: var(custom-property-variable, #fff)`

## Rule
`custom-property-pattern`

## Example
```css
// Fails lint rule
--invalid-custom-property-variable: #fff;

// Does not fail lint rule
a {
  color: --invalid-custom-property-variable;
}

// Does not fail lint rule
a {
  color: var(--invalid-custom-property-variable, #fff);
}
```

## Reproducible Config

```json
{
  rules: {
    'custom-property-pattern': [
      'foo-.+',
    ],
  },
}
```

## Version
^9.3.0

## Expected Behavior
The lint rule should be enforced when custom properties are used in a fallback syntax.

**Comment by StephenEsser** [comment]
Hi @jeddy3, thanks for getting back on this issue. 

I have no objections to that approach, the primary goal of this issue was to enable the lint rule for properties being substituted with the var function. 
```css
a {
  color: var(--invalid-custom-property-variable, #fff);
}
```
The false positive mentioned is not something we're immediately concerned with. It was just one of the ways I found that could get the lint rule to fail for an example.

**Comment by jeddy3** [comment]
@StephenEsser As there have been no objections to this proposal. Feel free to [contribute a pr](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) to address these false negatives. I suspect you'll want to make sure of `postcss-value-parser` to parse declaration values and check the pattern against `word`s that are the first child of `var` functions and begin with `--`.
